e2e_tests/utils/assertions.py

```diff
@@ -64,3 +67,34 @@ def format_violation(violation):
         violation_str += "\n"
 
     return violation_str
+
+
+def compare_snapshot(actual_path, expected_path):
+    actual_image = Image.open(actual_path).convert("RGB")
+    expected_image = Image.open(expected_path).convert("RGB")
+
+    diff = ImageChops.difference(actual_image, expected_image)
+    diff_np = np.array(diff)
+    num_diff_pixels = np.sum(np.any(diff_np != 0, axis=-1))
+
+    return num_diff_pixels <= 0
```

Comment:
The strictness of this check might cause issues across different hardware/browser versions